### PR TITLE
New unpacker integration

### PR DESCRIPTION
**Pull request by alekgajos**
This PR replaces the use of `Unpacker2` with the new unpacker from the [j-pet-unpacker](https://github.com/JPETTomography/j-pet-unpacker) repository used as a submodule.

The former `JPetUnpackTask` is now replaced by a new derivative of `JPetInputHandler`, `JPetInputHandlerHLD` which reads the HLD file using the new unpacker until data is exhausted and feeds the data to the 1st user task.

Formerly, `JPetInputHandler` was specific to ROOT data, therefore now it was transformed to a base class whereas the ROOT-specific code was moved to `JPetInputHandlerROOT`.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -45,3 +45,4 @@ build*
 
 compile_commands.json
 .vscode*
+.cache/*
```
Comment (wkrzemien) [inline .gitignore:48]: end of line is missing

File: include/Core/JPetTaskIO/JPetInputHandlerROOT.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  JPetTreeHeader* getHeaderClone() override;
+
+
+protected:
+  std::unique_ptr<JPetReaderInterface> fReader{nullptr};
+
+private:
+};
+
+#endif /*  !JPETINPUTHANDLERROOT_H */
```
Comment (wkrzemien) [inline include/Core/JPetTaskIO/JPetInputHandlerROOT.h:47]: end of line

File: src/Core/JPetTaskIO/JPetInputHandlerROOT.cpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  {
+    totalEntries = fReader->getNbOfAllEntries();
+  }
+  else
+  {
+    WARNING("no JPETReader,  totalEntries set to -1");
+    totalEntries = -1;
+  }
+  return JPetTaskIOTools::setUserLimits(options, totalEntries);
+}
```
Comment (wkrzemien) [inline src/Core/JPetTaskIO/JPetInputHandlerROOT.cpp:115]: end of line

File: src/Core/JPetTaskIO/JPetInputHandlerHLD.cpp [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    std::vector<uint32_t> corr_vec(150);
+    for (int i = 1; i < corr_histo->GetNbinsX(); ++i)
+    {
+      corr_vec[i - 1] = corr_histo->GetBinContent(i) * 1000.;
+    }
+
+    fTDCCalib[address][local_channel_no] = corr_vec;
+  }
+
+  // calib_rootfile->Close();
```
Comment (wkrzemien) [inline src/Core/JPetTaskIO/JPetInputHandlerHLD.cpp:179]: please, remove commented lines

**Comment by alekgajos** [comment]
Retest this please.

**Comment by alekgajos** [comment]
Retest this please

**Comment by JPETTomographyTester** [comment]
Found 12 violations:

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L517

Label &#39;foreach_preprocessed_data&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1356976337>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L483

Label &#39;foreach_filtered_data&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a850398026>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L419

Label &#39;foreach_original_data&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1890140265>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L379

Label &#39;foreach_endpoint&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1670827712>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L353

Label &#39;foreach_concentrator&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1719656846>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L319

Label &#39;loop_till_valid_queue&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-934766295>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: variableScope
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L149

The scope of the variable &#39;byte_cntr&#39; can be reduced. Warning: Be careful when fixing this message, especially when there are inner loops. Here is an example where cppcheck will write that the scope for &#39;i&#39; can be reduced:\012void f(int x)\012{\012    int i &#61; 0;\012    if (x) {\012        // it&#39;s safe to move &#39;int i &#61; 0;&#39; here\012        for (int n &#61; 0; n &lt; 10; ++n) {\012            // it is possible but not safe to move &#39;int i &#61; 0;&#39; here\012            do_something(&amp;i);\012        }\012    }\012}\012When you see this message it is always safe to reduce the variable scope 1 level.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1803094133>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unreadVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L106

Variable &#39;d&#39; is assigned a value that is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a2078580144>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unreadVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker_example.cpp L29

Variable &#39;paths_to_tdc_calib&#39; is assigned a value that is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-807912736>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker_example.cpp L11

Unused variable: data

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a4065189>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unmatchedSuppression
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/src/Core/JPetManager/JPetManager.cpp L121

Unmatched suppression: unusedFunction

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a709213197>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedFunction
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/src/Options/JPetOptionsTools/JPetOptionsTools.cpp L243

The function &#39;getUnpackerConfigFile&#39; is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-954319032>*

 *<ACCUMULATED-VIOLATIONS>*

**Comment by JPETTomographyTester** [comment]
Found 12 violations:

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L517

Label &#39;foreach_preprocessed_data&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-837883011>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L483

Label &#39;foreach_filtered_data&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-854122856>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L419

Label &#39;foreach_original_data&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a185619383>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L379

Label &#39;foreach_endpoint&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1748482318>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L353

Label &#39;foreach_concentrator&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a15135964>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L319

Label &#39;loop_till_valid_queue&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1781195803>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: variableScope
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L149

The scope of the variable &#39;byte_cntr&#39; can be reduced. Warning: Be careful when fixing this message, especially when there are inner loops. Here is an example where cppcheck will write that the scope for &#39;i&#39; can be reduced:\012void f(int x)\012{\012    int i &#61; 0;\012    if (x) {\012        // it&#39;s safe to move &#39;int i &#61; 0;&#39; here\012        for (int n &#61; 0; n &lt; 10; ++n) {\012            // it is possible but not safe to move &#39;int i &#61; 0;&#39; here\012            do_something(&amp;i);\012        }\012    }\012}\012When you see this message it is always safe to reduce the variable scope 1 level.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1398417177>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unreadVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L106

Variable &#39;d&#39; is assigned a value that is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-662804610>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unreadVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker_example.cpp L29

Variable &#39;paths_to_tdc_calib&#39; is assigned a value that is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a566825390>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker_example.cpp L11

Unused variable: data

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-317641357>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unmatchedSuppression
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/src/Core/JPetManager/JPetManager.cpp L121

Unmatched suppression: unusedFunction

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1140843429>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedFunction
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/src/Options/JPetOptionsTools/JPetOptionsTools.cpp L243

The function &#39;getUnpackerConfigFile&#39; is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-338017386>*

 *<ACCUMULATED-VIOLATIONS>*

**Comment by alekgajos** [comment]
Retest this please.

**Comment by JPETTomographyTester** [comment]
Found 12 violations:

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L517

Label &#39;foreach_preprocessed_data&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1356976337>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L483

Label &#39;foreach_filtered_data&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a850398026>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L419

Label &#39;foreach_original_data&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1890140265>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L379

Label &#39;foreach_endpoint&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1670827712>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L353

Label &#39;foreach_concentrator&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1719656846>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L319

Label &#39;loop_till_valid_queue&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-934766295>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: variableScope
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L149

The scope of the variable &#39;byte_cntr&#39; can be reduced. Warning: Be careful when fixing this message, especially when there are inner loops. Here is an example where cppcheck will write that the scope for &#39;i&#39; can be reduced:\012void f(int x)\012{\012    int i &#61; 0;\012    if (x) {\012        // it&#39;s safe to move &#39;int i &#61; 0;&#39; here\012        for (int n &#61; 0; n &lt; 10; ++n) {\012            // it is possible but not safe to move &#39;int i &#61; 0;&#39; here\012            do_something(&amp;i);\012        }\012    }\012}\012When you see this message it is always safe to reduce the variable scope 1 level.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1803094133>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unreadVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L106

Variable &#39;d&#39; is assigned a value that is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a2078580144>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unreadVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker_example.cpp L29

Variable &#39;paths_to_tdc_calib&#39; is assigned a value that is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-807912736>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker_example.cpp L11

Unused variable: data

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a4065189>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unmatchedSuppression
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/src/Core/JPetManager/JPetManager.cpp L121

Unmatched suppression: unusedFunction

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a709213197>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedFunction
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/src/Options/JPetOptionsTools/JPetOptionsTools.cpp L243

The function &#39;getUnpackerConfigFile&#39; is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-954319032>*

 *<ACCUMULATED-VIOLATIONS>*

**Comment by JPETTomographyTester** [comment]
Found 12 violations:

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L517

Label &#39;foreach_preprocessed_data&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-837883011>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L483

Label &#39;foreach_filtered_data&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-854122856>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L419

Label &#39;foreach_original_data&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a185619383>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L379

Label &#39;foreach_endpoint&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1748482318>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L353

Label &#39;foreach_concentrator&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a15135964>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedLabel
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L319

Label &#39;loop_till_valid_queue&#39; is not used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1781195803>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: variableScope
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L149

The scope of the variable &#39;byte_cntr&#39; can be reduced. Warning: Be careful when fixing this message, especially when there are inner loops. Here is an example where cppcheck will write that the scope for &#39;i&#39; can be reduced:\012void f(int x)\012{\012    int i &#61; 0;\012    if (x) {\012        // it&#39;s safe to move &#39;int i &#61; 0;&#39; here\012        for (int n &#61; 0; n &lt; 10; ++n) {\012            // it is possible but not safe to move &#39;int i &#61; 0;&#39; here\012            do_something(&amp;i);\012        }\012    }\012}\012When you see this message it is always safe to reduce the variable scope 1 level.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1398417177>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unreadVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker.hpp L106

Variable &#39;d&#39; is assigned a value that is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-662804610>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unreadVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker_example.cpp L29

Variable &#39;paths_to_tdc_calib&#39; is assigned a value that is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a566825390>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/j-pet-unpacker/unpacker_example.cpp L11

Unused variable: data

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-317641357>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unmatchedSuppression
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/src/Core/JPetManager/JPetManager.cpp L121

Unmatched suppression: unusedFunction

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1140843429>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedFunction
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake351/gcc/default_gcc/os/ubuntu16/root/root612/src/Options/JPetOptionsTools/JPetOptionsTools.cpp L243

The function &#39;getUnpackerConfigFile&#39; is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-338017386>*

 *<ACCUMULATED-VIOLATIONS>*

**Comment by JPETTomographyTester** [comment]
Found 4 violations:

**Reporter**: cppcheck-matrix-raporter
**Rule**: unreadVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker.hpp L107

Variable &#39;d&#39; is assigned a value that is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1226406485>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unreadVariable
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/j-pet-unpacker/unpacker_example.cpp L27

Variable &#39;paths_to_tdc_calib&#39; is assigned a value that is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a139345876>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unmatchedSuppression
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/src/Core/JPetManager/JPetManager.cpp L121

Unmatched suppression: unusedFunction

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a577067174>*

**Reporter**: cppcheck-matrix-raporter
**Rule**: unusedFunction
**Severity**: INFO
**File**: /var/jenkins_home/workspace/J-Pet-Framework-matrix-job/boost_version/boost_1_58/cmake/cmake315/gcc/default_gcc/os/ubuntu16/root/root618/src/Options/JPetOptionsTools/JPetOptionsTools.cpp L243

The function &#39;getUnpackerConfigFile&#39; is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1693352471>*

 *<ACCUMULATED-VIOLATIONS>*

**Comment by alekgajos** [comment]
All tests are passing and CppCheck-suggested fixes are applied (and sometimes suppressed where appropriate).

@wkrzemien Please have a look.

**Review by wkrzemien — APPROVED** [review]
Looks nice!

**Pull request merged**